src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/Target_ARM/ARMJumpStubNode.cs

```diff
@@ -11,7 +11,8 @@ protected override void EmitCode(NodeFactory factory, ref ARMEmitter encoder, bo
         {
             if (!_target.RepresentsIndirectionCell)
             {
-                encoder.EmitJMP(_target); // b methodEntryPoint
+                int delta = (_target is IMethodNode || _target is AssemblyStubNode) ? factory.Target.CodeDelta : 0;
+                encoder.EmitJMP(_target, delta); // b methodEntryPoint
```

Comment:
A jump is always to code, so we want to always pass the code delta.

```suggestion
                encoder.EmitJMP(_target, factory.Target.CodeDelta); // b methodEntryPoint
```